src/Config/MergeConfigProvider.cs

```diff
@@ -106,10 +108,11 @@ private static void MergeObjects(Utf8JsonWriter jsonWriter, JsonElement root1, J
         }
 
         /// <summary>
-        /// This helper methods tries to merge two Json Array such that the values of second array
-        /// completely overrides that of the first.
+        /// This helper methods completely overrides the elements from the first config with that of second.
+        /// For example: For an entity, The permissions defined in the override config would completely replace
```

Comment:
> in the override config

following my other comments, this is ambiguous at first glace consider the presence of these three files:
1. dab-config.json
2. dab-config.mssql.json
3. dab-config.mssql.overrides.json 

When the term override is used here, you are not referring to number 3 above

Reply:
i see your concerns. will think of some other name

Reply:
Just to add the final merged config is called `dab-config.{DAB_ENVIRONMENT}.merged.json`